circuit/pallets/execution-delivery/src/lib.rs

```diff
@@ -457,9 +192,40 @@ pub mod pallet {
         }
     }
 
-    /// A public part of the pallet.
     #[pallet::call]
     impl<T: Config> Pallet<T> {
+        #[pallet::weight(<T as Config>::WeightInfo::dry_run_whole_xtx_three_components() + <T as Config>::WeightInfo::decompose_io_schedule())]
+        pub fn submit_exec(
+            origin: OriginFor<T>,
+            contract_id: RegistryContractId<T>,
+            input: Vec<u8>,
+            value: BalanceOf<T>,
+            reward: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
```

Comment:
Events + SideEffects are emitted by `submit_xtx_execution` which I encapsulated as a separate method since most of the logic is shared with `submit_composable_exec_order`. And yep, the docs need to be updated in general